### PR TITLE
Add tip about Comment Block style within JSX

### PR DESCRIPTION
**Pull request by thadguidry**
## Pre-flight checklist

- [x] I have read the [Contributing Guidelines on pull requests](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#pull-requests).
- [ ] **If this is a code change**: I have written unit tests and/or added dogfooding pages to fully verify the new behavior.
- [ ] **If this is a new API or substantial change**: the PR has an accompanying issue (closes #0000) and the maintainers have approved on my working plan.

## Motivation

- Adds tip admonition about JSX comment blocks just before we introduce some info about Prettier and using a JSX comment block
- This is also useful context later in the page where we talk about React component styling a bit.
- A reply to this PR might be - "we don't want to teach everything about JSX!!" which is true, but definitely about comment blocks in JSX within this page would be very useful and also improves searchability about "comment blocks".

### Test links

Deploy preview: https://deploy-preview-10066--docusaurus-2.netlify.app/

### CODE REVIEW

**Review by Josh-Cena — CHANGES_REQUESTED** [review]
This should be put in the markdown features file instead. This doc is for how MDX and React interop (importing/exporting components, etc.)

File: website/docs/guides/markdown-features/markdown-features-intro.mdx [hunk 1]
```diff
@@ -220,3 +220,33 @@ Markdown can embed HTML elements, and [`details`](https://developer.mozilla.org/
...
+    color: 'pink'
+  }}>
+    <li>Improve</li>
+    <li>Prepare</li>
+    {/* Our CTO preferred Work, but we might change this to Play in the future */}
+    <li>Work</li>
+  </ul>
+
+</BrowserWindow>
+```
```
Comment (slorber) [inline website/docs/guides/markdown-features/markdown-features-intro.mdx:252]: This is too verbose for me just to explain we should use `{/* comment */}` syntax in MDX. Even the MDX docs is less verbose than this when explaining comments: https://mdxjs.com/docs/what-is-mdx/



This does does not take into account that we also have a commonmark mode that still allows to use HTML comment. 

We also have a v3 compat	option to support HTML comments in MDX that is turned on by default: https://docusaurus.io/docs/migration/v3#comments-option

**Review by slorber — COMMENTED** [review]
> we don't want to teach everything about JSX

Yes 👍 there are many other JSX things that we don't mention here, and we already link to the MDX docs for those that want to know more.

---

Thanks for your proposal, and I'm not against mentioning comments, but if you care about this addition, please open a real PR (not from a patch-1 branch), ensure it passes linting, and keep the information as concise as possible, and 100% accurate.

**Comment by thadguidry** [comment]
@slorber no worries.  Until next time! :-)

**Pull request closed without merging**